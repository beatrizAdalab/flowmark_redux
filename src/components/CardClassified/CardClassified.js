import React from 'react';
import { Link } from 'react-router-dom';
import { LoginConsumer } from '../../context/LoginContext';
import ReactImageFallback from 'react-image-fallback';
import nophoto from '../../assets/images/nophoto.png';

// get our fontawesome imports
import { faPencilAlt } from '@fortawesome/free-solid-svg-icons';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';


const CardClassified = ({ classified }) => {
    return (
        <LoginConsumer>
            {(value) => {

                return (
                    <div className='col-12 col-md-4 col-lg-4 mb-5'>
                        <div className='card shadow-sm bg-white rounded d-flex flex-column justify-content-between h-100'>
                            <div>
                                <Link
                                    to={`/detailClassifieds/${classified._id}`}
                                    key={classified.id}
                                    className='text-decoration-none text-body'
                                >
                                    <div className='card-container-img card-image '>
                                        <ReactImageFallback
                                            src={classified.photo}
                                            fallbackImage={nophoto}
                                            className='card-img card-image border-bottom'
                                            alt={classified.name}
                                        />
                                    </div>
                                    <div className=''>
                                        <span className='m-2 badge badge-danger'>{classified.price} â‚¬</span>
                                    </div>
                                    <div className='card-body p-2'>
                                        <h5 className='card-title'> {classified.name}</h5>
                                        <p > {classified.description} </p>
                                        <div>
                                            {classified.tags ?
                                                classified.tags.map((tag, index) => <span key={index} className='badge badge-pill mr-2 mb-2 badge-secondary'> {tag} </span>)
                                                : false}
                                        </div>
                                    </div>
                                </Link>
                            </div>
                            <div className='card-footer bg-transparent p-2'>
                                <Link
                                    to={`/editClassifieds/${classified._id}`}
                                    className='text-decoration-none text-body'
                                >
                                    <FontAwesomeIcon className='mr-2 text-edit' icon={faPencilAlt} />
                                    Edit
                            </Link>
                            </div>
                        </div>
                    </div>
                );
            }}
        </LoginConsumer>
    );
};

export default CardClassified;