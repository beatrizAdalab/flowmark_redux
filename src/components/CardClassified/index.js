import CardClassified from './CardClassified';

export default CardClassified;