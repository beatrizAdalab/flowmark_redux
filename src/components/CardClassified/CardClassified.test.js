import React from 'react';
import { shallow } from 'enzyme';
import CardClassified from './CardClassified';

describe('CardClassified', () => {
  const props = {
    classified: {
      tags: ["lifestyle", "work", "mobile"],
      _id: "5e43096e99660405061db854",
      name: "Primero 4",
      price: 10000,
      description: "cambio",
      type: "sell",
      photo: "https://pbs.twimg.com/media/Dpu128FWwAI_Bru?format=jpg",
      createdAt: "2020-02-11T20:07:10.232Z",
      updatedAt: "2020-05-28T14:46:59.662Z",
      __v: 15
    }
  };
  let wrapper;

  test('snapshot testing', () => {
    expect(wrapper).toMatchSnapshot();
  });

  beforeEach(() => {
    wrapper = shallow(<CardClassified {...props} />);
  });

  test('should render', () => {
    expect(wrapper.exists()).toBe(true);
  });

  test('should have a photo in component ReactImageFallback', () => {
    expect(wrapper.find('ReactImageFallback').props().src).toBe(props.classified.photo)
  });

});