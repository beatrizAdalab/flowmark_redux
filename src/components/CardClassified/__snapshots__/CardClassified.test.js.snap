// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardClassified snapshot testing 1`] = `
<div
  className="col-12 col-md-4 col-lg-4 mb-5"
>
  <div
    className="card shadow-sm bg-white rounded d-flex flex-column justify-content-between h-100"
  >
    <div>
      <Link
        className="text-decoration-none text-body"
        to="/detailClassifieds/5e43096e99660405061db854"
      >
        <div
          className="card-container-img card-image "
        >
          <ReactImageFallback
            alt="Primero 4"
            className="card-img card-image border-bottom"
            fallbackImage="nophoto.png"
            initialImage={null}
            src="https://pbs.twimg.com/media/Dpu128FWwAI_Bru?format=jpg"
          />
        </div>
        <div
          className=""
        >
          <span
            className="m-2 badge badge-danger"
          >
            10000
             â‚¬
          </span>
        </div>
        <div
          className="card-body p-2"
        >
          <h5
            className="card-title"
          >
             
            Primero 4
          </h5>
          <p>
             
            cambio
             
          </p>
          <div>
            <span
              className="badge badge-pill mr-2 mb-2 badge-secondary"
              key="0"
            >
               
              lifestyle
               
            </span>
            <span
              className="badge badge-pill mr-2 mb-2 badge-secondary"
              key="1"
            >
               
              work
               
            </span>
            <span
              className="badge badge-pill mr-2 mb-2 badge-secondary"
              key="2"
            >
               
              mobile
               
            </span>
          </div>
        </div>
      </Link>
    </div>
    <div
      className="card-footer bg-transparent p-2"
    >
      <Link
        className="text-decoration-none text-body"
        to="/editClassifieds/5e43096e99660405061db854"
      >
        <FontAwesomeIcon
          border={false}
          className="mr-2 text-edit"
          fixedWidth={false}
          flip={null}
          icon={
            Object {
              "icon": Array [
                512,
                512,
                Array [],
                "f303",
                "M497.9 142.1l-46.1 46.1c-4.7 4.7-12.3 4.7-17 0l-111-111c-4.7-4.7-4.7-12.3 0-17l46.1-46.1c18.7-18.7 49.1-18.7 67.9 0l60.1 60.1c18.8 18.7 18.8 49.1 0 67.9zM284.2 99.8L21.6 362.4.4 483.9c-2.9 16.4 11.4 30.6 27.8 27.8l121.5-21.3 262.6-262.6c4.7-4.7 4.7-12.3 0-17l-111-111c-4.8-4.7-12.4-4.7-17.1 0zM124.1 339.9c-5.5-5.5-5.5-14.3 0-19.8l154-154c5.5-5.5 14.3-5.5 19.8 0s5.5 14.3 0 19.8l-154 154c-5.5 5.5-14.3 5.5-19.8 0zM88 424h48v36.3l-64.5 11.3-31.1-31.1L51.7 376H88v48z",
              ],
              "iconName": "pencil-alt",
              "prefix": "fas",
            }
          }
          inverse={false}
          listItem={false}
          mask={null}
          pull={null}
          pulse={false}
          rotation={null}
          size={null}
          spin={false}
          swapOpacity={false}
          symbol={false}
          title=""
          transform={null}
        />
        Edit
      </Link>
    </div>
  </div>
</div>
`;
